src/liger_kernel/ops/utils.py

```diff
@@ -41,6 +42,96 @@ def calculate_settings(n):
     return BLOCK_SIZE, num_warps
 
 
+def calculate_gemm_settings(m, n, k):
+    total_memory = torch.cuda.get_device_properties(0).total_memory
+
+    if total_memory >= 48 * 1000 * 1000 * 1000:  # >=48 GB VRAM
+        if m * n * k > 1e9:  # large matmul
+            return (
+                128,  # block_m
+                128,  # block_n
+                512,  # block_k
+                4,  # num_stages
+                8,  # num_warps
+                4,  # split_k
+                16,  # group_m
+            )
+        elif m * n * k > 1e6:  # mid matmul
+            return (
+                128,  # block_m
+                128,  # block_n
+                256,  # block_k
+                3,  # num_stages
+                8,  # num_warps
+                2,  # split_k
+                8,  # group_m
+            )
+        else:  # small matmul
+            return (
+                64,  # block_m
+                64,  # block_n
+                128,  # block_k
+                3,  # num_stages
+                4,  # num_warps
+                1,  # split_k
+                4,  # group_m
+            )
+    else:  # <48 GB VRAM
+        if m * n * k > 1e9:  # large matmul
+            return (
+                64,  # block_m
+                64,  # block_n
+                128,  # block_k
+                3,  # num_stages
+                4,  # num_warps
+                2,  # split_k
+                8,  # group_m
+            )
+        elif m * n * k > 1e4:  # mid matmul
+            return (
+                64,  # block_m
+                64,  # block_n
+                64,  # block_k
+                3,  # num_stages
+                4,  # num_warps
+                1,  # split_k
+                4,  # group_m
+            )
+        else:  # small matmul
+            return (
+                64,  # block_m
+                64,  # block_n
+                64,  # block_k
+                3,  # num_stages
+                4,  # num_warps
+                1,  # split_k
+                2,  # group_m
+            )
+
+
+def check_compute_capability_for_fp8(fn):
+    @functools.wraps(fn)
+    def wrapper(ctx, *args, **kwargs):
+        if torch.cuda.is_available():
+            device = torch.cuda.current_device()
+            compute_capability = torch.cuda.get_device_capability(device)
+            if compute_capability[0] >= 9:  # SM_90+
+                os.environ["ENABLE_TMA"] = "1"
```

Comment:
i'm wondering if we should care about `ENABLE_MMA_V3` too 🤔

Reply:
Yeah I think that's a good idea. The H100 has wgmma units which has much higher throughput than previous generations.